spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkBatch.java

```diff
@@ -137,10 +137,24 @@ public PartitionReaderFactory createReaderFactory() {
   }
 
   private ParquetBatchReadConf parquetBatchReadConf(ParquetReaderType readerType) {
-    return ImmutableParquetBatchReadConf.builder()
-        .batchSize(readConf.parquetBatchSize())
-        .readerType(readerType)
-        .build();
+    String factoryClassName = readConf.parquetVectorizedReaderFactory();
+
+    // If no explicit factory is set and reader type is COMET, use the default Comet factory
+    if (factoryClassName == null && readerType == ParquetReaderType.COMET) {
```

Comment:
This can be handled inside `readConf.parquetVectorizedReaderFactory()` and moved to right before the check `if (factoryClassName != null) {`.